spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalogManager.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.spark.catalog;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.client.GravitinoClient;
+import com.datastrato.gravitino.client.GravitinoMetaLake;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** GravitinoCatalogManager is used to sync catalogs from Gravitino server. */
+public class GravitinoCatalogManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoCatalogManager.class);
+  private static GravitinoCatalogManager gravitinoCatalogManager;
+
+  private volatile boolean isClosed = false;
+  private HashMap<String, Catalog> gravitinoCatalogs = new HashMap<>();
+  private String metalakeName;
+  private GravitinoClient gravitinoClient;
+
+  private GravitinoCatalogManager(String gravitinoUri, String metalakeName) {
+    this.metalakeName = metalakeName;
+    this.gravitinoClient = GravitinoClient.builder(gravitinoUri).build();
+  }
+
+  public static synchronized GravitinoCatalogManager createGravitinoCatalogManager(
+      String gravitinoUrl, String metalakeName) {
+    if (gravitinoCatalogManager != null) {
+      throw new RuntimeException("should not create duplicate GravitinoCatalogManager");
+    }
+    gravitinoCatalogManager = new GravitinoCatalogManager(gravitinoUrl, metalakeName);
+    return gravitinoCatalogManager;
+  }
+
+  public static synchronized GravitinoCatalogManager getGravitinoCatalogManager() {
+    if (gravitinoCatalogManager == null) {
+      throw new RuntimeException("GravitinoCatalogManager has not created yet");
+    }
+    if (gravitinoCatalogManager.isClosed) {
+      throw new RuntimeException("GravitinoCatalogManager " + "is already closed");
+    }
+    return gravitinoCatalogManager;
+  }
+
+  public void close() {
+    Preconditions.checkArgument(isClosed == false);
+    isClosed = true;
+    gravitinoClient.close();
+  }
+
+  public Catalog getGravitinoCatalogInfo(String name) {
+    return gravitinoCatalogs.get(name);
+  }
+
+  public Map<String, Catalog> getGravitinoCatalogs() {
+    return ImmutableMap.copyOf(gravitinoCatalogs);
+  }
+
+  public String getMetalakeName() {
+    return metalakeName;
+  }
+
+  public void loadCatalogsFromGravitino() {
+    LOG.info("Load metalake: " + metalakeName);
```

Comment:
This is not so meaningful as only for debugging.

Reply:
removed